src/data/repositories/UserD2ApiRepository.ts

```diff
@@ -51,6 +51,8 @@ export class UserD2ApiRepository implements UserRepository {
     }
 
     private mapUser(user: D2ApiUser): User {
+        const lastLogin = user.userCredentials.lastLogin;
```

Comment:
Unless we re-use/transform a variable we normally use them directly instead of copying them to a local variable.